crates/neon-macros/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+//! Procedural macros supporting [Neon](https://docs.rs/neon/latest/neon/)
+
+#[cfg(feature = "napi")]
+mod napi;
+#[cfg(feature = "napi")]
+use napi as macros;
+
+#[cfg(not(feature = "napi"))]
+mod legacy;
+#[cfg(not(feature = "napi"))]
+use legacy as macros;
+
+// Proc macro definitions must be in the root of the crate
+// Implementations are in the backend dependent module
+
+#[proc_macro_attribute]
+/// Marks a method as the primary entrypoint for initialization in a Neon
+/// module. This attribute should only be used _once_ in a module and will
```

Comment:
State can only be shared within a _single_ crate. Do you think the complexity is worth it for a clearer error message when we can only catch multiple uses _within_ a crate and not across crates?

Using this macro multiple times is a compile time error for N-API and a runtime error for legacy.

```
error: symbol `napi_register_module_v1` is already defined
  --> test/napi/native/src/js/types.rs:63:1
   |
63 | #[neon::init]
   | ^^^^^^^^^^^^^
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

https://github.com/rust-lang/rust/issues/44034#issuecomment-371631945

Reply:
We could require that users actually call the function `init`. You could technically still have multiple `init` functions in different modules, but…

Reply:
For comparison, the exact same bug exists with the current macro and I can't think of a single bug report we've ever had because of that issue. With that said, a proc_macro hides a bit more than the big normal macro.

How would we ensure that `neon::init` is used at least once?